test/DotNetEnv.Tests/ParserTests.cs

```diff
@@ -361,6 +387,21 @@ public void ParseAssignment ()
             Assert.Throws<ParseException>(() => Parsers.Assignment.End().Parse("EV_DNE='"));
             Assert.Throws<ParseException>(() => Parsers.Assignment.End().Parse("EV_DNE=0\n1"));
 
+            testParse("EV_DNE", "a'b'' 'c' d", "EV_DNE=\"a'b'' 'c' d\" #allow singleQuotes in doubleQuoted values");
+            testParse("EV_DNE", "a\"b\"\" \"c\" d", "EV_DNE='a\"b\"\" \"c\" d' #allow doubleQuotes in singleQuoted values");
+
+            testParse("EV_DNE", "", "EV_DNE=");
+            testParse("EV_DNE", "EV_DNE=", "EV_DNE=EV_DNE=");
+
+            testParse("EV_DNE", "test", "EV_DNE= test #basic comment");
+            testParse("EV_DNE", "", "EV_DNE=#no value just comment");
+            testParse("EV_DNE", "", "EV_DNE= #no value just comment");
+            testParse("EV_DNE", "test", "EV_DNE= test  #a'bc allow singleQuotes in comment");
+            testParse("EV_DNE", "test", "EV_DNE= test  #a\"bc allow doubleQuotes in comment");
+            testParse("EV_DNE", "test", "EV_DNE= test #a$bc allow dollarSign in comment");
+
+            testParse("EV_DNE", "http://www.google.com/#anchor", "EV_DNE=http://www.google.com/#anchor #inline hash is part of value");
+
```

Comment:
Please just double check that we both added all the same tests, in more or less all the same places.

Reply:
- ParseUnquotedValue
  - reordered and at some tests removed the "End", because it is not needed (and we expect the same result in both cases, but with the End we cover only the End, while without we expect it to throw if it reads until the end or not, it doesn't matter)
- ParseValue
  - the "whitespace throws"-Test was missing, added it; (without "End" again because it wasn't necessary to have it throwing)
  - remove the "duplicated" unmatched quote-test with the "End"-statement. That is obsolete because it doesn't test additional actual behaviour.
- ParseAssignment
  - added `testParse("EV_DNE", "a#b#c", "EV_DNE=a#b#c # comment");` (changed the comment to be more explaining)
  - renamed "inline hash" to "inner hash", according to your comment about "inline quotes", which is also not real inline, but inner
  - there are 4 tests with a comment "already working"; they are duplicates compared to these two tests:
    - `testParse("EV_DNE", "a'b'' 'c' d", "EV_DNE=\"a'b'' 'c' d\" #allow singleQuotes in doubleQuoted values");`
    - `testParse("EV_DNE", "a\"b\"\" \"c\" d", "EV_DNE='a\"b\"\" \"c\" d' #allow doubleQuotes in singleQuoted values");`
  - added `testParse("EV_DNE", "a\"b\"\" \"c\" d", "EV_DNE=\"a\\\"b\\\"\\\" \\\"c\\\" d\" #allow quoted doubleQuotes in doubleQuoted values");` (quite good test, did not see this one before; fixed the comment tho, should be "escaped doubleQuotes" instead of "quoted doubleQuotes")
  - added `Assert.Throws<ParseException>(() => Parsers.Assignment.End().Parse("EV_DNE='a'b'' 'c' d'"));  // no single quotes inside single quoted values`
    - added an equivalent test for (unescaped) doubleQuotes inside doubleQuoted too
    - removed the "End", not necessary to let it throw here too ;)
  - added `testParse("EV_DNE", "a#b#c# not a comment", "EV_DNE=a#b#c# not a comment");`
  - I did not add the the huge comment and the corresponding tests for inner quotes in unquoted values; imho this is well tested in `ParseUnquotedValue` directly; that's also true for quite some other things, many stuff in assignment is duplicated from other methods; but due to the huge comment I decided to not adding this --> open for other thoughts ;)

That should be all, pushing it in a few seconds, wrote that while changing the code accordingly ;)

Reply:
Sounds good, thanks!